inputfiles/overridingTypes.json

```diff
@@ -2400,6 +2455,243 @@
+                }
+            },
+            "ReadableStreamReadResult": {
+                "type-parameters": [
+                    {
+                        "name": "T"
+                    }
+                ],
+                "properties": {
+                    "property": {
+                        "value": {
+                            "override-type": "T"
+                        }
+                    }
+                },
+                "no-interface-object": "1"
+            },
+            "ReadableStreamBYOBReader": {
+                "constructor": {
+                    "override-signatures": [
+                        "new(stream: ReadableStream<Uint8Array>): ReadableStreamBYOBReader"
+                    ]
+                },
+                "methods": {
+                    "method": {
+                        "read": {
+                            "override-signatures": [
+                                "read<T extends ArrayBufferView>(view: T): Promise<ReadableStreamReadResult<T>>"
+                            ]
+                        }
+                    }
+                }
+            },
+            "ReadableStreamDefaultReader": {
+                "type-parameters": [
+                    {
+                        "name": "R",
+                        "default": "any"
+                    }
+                ],
+                "constructor": {
+                    "override-signatures": [
+                        "new<R = any>(stream: ReadableStream<R>): ReadableStreamDefaultReader<R>"
+                    ]
+                },
+                "methods": {
+                    "method": {
+                        "read": {
+                            "override-signatures": [
+                                "read(): Promise<ReadableStreamReadResult<R>>"
+                            ]
+                        }
+                    }
+                }
+            },
+            "ReadableByteStreamController": {
+                "properties": {
+                    "property": {
+                        "byobRequest": {
+                            "override-type": "ReadableStreamBYOBRequest | undefined"
+                        }
+                    }
+                }
+            },
+            "ReadableStreamDefaultController": {
+                "type-parameters": [
+                    {
+                        "name": "R",
+                        "default": "any"
+                    }
+                ],
+                "methods": {
+                    "method": {
+                        "enqueue": {
+                            "override-signatures": [
+                                "enqueue(chunk: R): void"
+                            ]
+                        }
+                    }
+                }
+            },
+            "ReadableWritableStreamPair": {
+                "name": "ReadableWritableStreamPair<R extends ReadableStream, W extends WritableStream>",
```

Comment:
Right now, `TypeParameter` only supports `name` and `default`. Therefore, I simply patched in the type parameters into directly the interface's name. Ideally, this would go into the `"type-parameters"` property of the interface.

I tried to fix this by adding an `extends` property, but that immediately caused the emitter to throw a `"Maximum call stack size exceeded"` error in `getParentsWithEventHandler` and `getParentEventHandler`. I'll look into this again later, to see why this would cause infinite recursion.

Reply:
Found it! The overriding interface still needs a `name` property for this to work, which I had removed while testing. Putting the `name` back fixes it.